lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,609 @@
+	
+	return util.TraceEntity( data, ply )
+end
+
+--[[
+Attemps to place the player at this position or as close as possible.
+]]--
+-- Directions to check
+local directions = {
+	Vector(0,0,0), Vector(0,0,1), --Center and up
+	Vector(1,0,0), Vector(-1,0,0), Vector(0,1,0), Vector(0,-1,0) --All cardinals
+}
+
+for deg=45,315,90 do -- Diagonals
+	local r = math.rad(deg)
+	table.insert(directions, Vector(math.Round(math.cos(r)), math.Round(math.sin(r)), 0))
+end
+
+local magn = 15 -- How much increment for each iteration
+local iterations = 2 -- How many iterations
+function NLL.PlayerSetPosNoBlock( ply, pos, filter )
+	local tr
+	
+	local dirvec
+	local m = magn
+	local i = 1
+	local its = 1
+	repeat
+		dirvec = directions[i] * m
+		i = i + 1
+		if i > #directions then
+			its = its + 1
+			i = 1
+			m = m + magn
+			if its > iterations then
+				ply:SetPos(pos) -- We've done as many checks as we wanted, lets just force him to get stuck then.
+				return false
+			end
+		end
+		
+		tr = NLL.PlayerHullTrace(dirvec + pos, ply, filter)
+	until tr.Hit == false
+	
+	ply:SetPos(pos + dirvec)
+	return true
+end
+
+
+--[[
+Sets the player invisible/visible
+]]--
+function NLL.PlayerInvis( ply, bool )
+	ply:SetNoDraw(bool)
+	ply:DrawShadow(not bool)
+
+	if pac and pac.TogglePartDrawing then
+		pac.TogglePartDrawing(ply, not bool)
+	end
+end
+
+
+--[[
+Deploy player ragdoll, Sleeping Variants can be woken up with E.
+]]--
+function NLL.Ragdoll( ply, pushdir, stuntype)
+	if not ply:Alive() or IsValid(ply.nllragdoll) then return end -- you already have a body
+	local plyphys = ply:GetPhysicsObject()
+	local plyvel = Vector(0,0,0)
+	if plyphys:IsValid() then
+		plyvel = plyphys:GetVelocity()
+	end
+	
+	ply.nlldroppos = ply:GetPos() -- Store pos incase the ragdoll is missing when we're to unrag him.
+	
+	local rag = ents.Create("prop_ragdoll")
+		rag:SetModel(ply:GetModel())
+		rag:SetPos(ply:GetPos())
+		rag:SetAngles(Angle(0,ply:GetAngles().y,0))
+		rag:SetColor(ply:GetColor())
+		rag:SetMaterial(ply:GetMaterial())
+		rag:Spawn()
+		rag:Activate()
+	
+	if not IsValid(rag:GetPhysicsObject()) then
+		MsgN("A slept player didn't get a valid ragdoll. Model ("..ply:GetModel()..")!")
+		SafeRemoveEntity(rag)
+		return false
+	end
+	
+	--Lower inertia makes the ragdoll have trouble rolling. Citizens have 1,1,1 as default, while combines have 0.2,0.2,0.2.
+	rag:GetPhysicsObject():SetInertia(Vector(1,1,1)) 
+	
+	--Push him back abit
+	plyvel = plyvel + pushdir*200
+	rag:GetPhysicsObject():SetVelocity(plyvel)
+		
+	--Stop firing of weapons
+	NLL.PlyStoreWeapons(ply)
+	ply:StripWeapons()
+	
+	--Makes him not collide with anything, including traces.
+	ply:SetCollisionGroup(COLLISION_GROUP_IN_VEHICLE)
+	
+	--Make him follow the ragdoll, if the player gets away from the ragdoll he won't get stuff rendered properly.
+	ply:SetParent(rag)
+	
+	--Make the player invisible.
+	NLL.PlayerInvis(ply, true)
+
+	ply.nllragdoll = rag
+	rag.nllplayer = ply
+	ply.condition = stuntype
```

Comment:
`ply.condition` and `NLL.ConditionStorage` achieve the same thing yet it's stored separately twice and used differently to achieve the same result. Get rid of one of them and make the rest of the code use the same method.